src/library/clementine/clementineplaylistmodel.cpp

```diff
@@ -0,0 +1,264 @@
+            query.bindValue(":" CLM_BITRATE, entry.bitrate);
+            query.bindValue(":" CLM_COMMENT, entry.comment);
+            query.bindValue(":" CLM_PLAYCOUNT, entry.playcount);
+            query.bindValue(":" CLM_COMPOSER, entry.composer);
+
+            if (!query.exec()) {
+                LOG_FAILED_QUERY(query);
+            }
+        }
+        endInsertRows();
+    }
+
+    QStringList tableColumns;
+    tableColumns << CLM_VIEW_ORDER // 0
+                 << CLM_PREVIEW;
+
+    QStringList trackSourceColumns;
+    trackSourceColumns << CLM_VIEW_ORDER // 0
+                       << CLM_ARTIST
+                       << CLM_TITLE
+                       << CLM_DURATION
+                       << CLM_URI
+                       << CLM_ALBUM
+                       << CLM_ALBUM_ARTIST
+                       << CLM_YEAR
+                       << CLM_RATING
+                       << CLM_GENRE
+                       << CLM_GROUPING
+                       << CLM_TRACKNUMBER
+                       << CLM_BPM
+                       << CLM_BITRATE
+                       << CLM_COMMENT
+                       << CLM_PLAYCOUNT
+                       << CLM_COMPOSER;
+
+    QSharedPointer<BaseTrackCache> trackSource(
+            new BaseTrackCache(m_pTrackCollectionManager->internalCollection(),
+                    Clementine_TABLE,
+                    CLM_VIEW_ORDER,
+                    trackSourceColumns,
+                    false));
+
+    setTable(Clementine_TABLE, CLM_VIEW_ORDER, tableColumns, trackSource);
+    setSearch("");
+    setDefaultSort(fieldIndex(PLAYLISTTRACKSTABLE_POSITION), Qt::AscendingOrder);
+    setSort(defaultSortColumn(), defaultSortOrder());
+}
+
+TrackModel::Capabilities ClementinePlaylistModel::getCapabilities() const {
+    return Capability::AddToTrackSet |
+            Capability::AddToAutoDJ |
+            Capability::LoadToDeck |
+            Capability::LoadToSampler;
+}
+
+Qt::ItemFlags ClementinePlaylistModel::flags(const QModelIndex& index) const {
+    return readWriteFlags(index);
+}
+
+TrackId ClementinePlaylistModel::doGetTrackId(const TrackPointer& pTrack) const {
+    if (pTrack) {
+        for (int row = 0; row < rowCount(); ++row) {
+            const QUrl rowUrl(getFieldString(index(row, 0), CLM_URI));
+            if (TrackFile::fromUrl(rowUrl) == pTrack->getFileInfo()) {
+                return TrackId(getFieldVariant(index(row, 0), CLM_VIEW_ORDER));
+            }
+        }
+    }
+    return TrackId();
+}
+
+QVariant ClementinePlaylistModel::getFieldVariant(const QModelIndex& index,
+        const QString& fieldName) const {
+    return index.sibling(index.row(), fieldIndex(fieldName)).data();
+}
+
+QString ClementinePlaylistModel::getFieldString(const QModelIndex& index,
+        const QString& fieldName) const {
+    return getFieldVariant(index, fieldName).toString();
+}
+
+TrackPointer ClementinePlaylistModel::getTrack(const QModelIndex& index) const {
```

Comment:
Why don't you rely on the existing implementation in BaseExternalPlaylistModel::getTrack() and instead decided to override it?

I guess the reason is that you used the uncommon Banshee integration as your blueprint instead of the more recent iTunes/Traktor/Rekordbox integrations? I suggest to align your implementation with those.

Reply:
i will try

Reply:
I am fine with the current solution. 
There is not a question of more recent. The Banshee integration is more recent than the idea of importing everything into a temporary mixxx db table. The Rekordbox integration is more recent. In terms of refactoring the Banshee code is a bit behind.

Reply:
okay if this is okay so far, i will not change it anymore.